File: cs/ch05/Money/Franc.cs
namespace TheSoftwareGorilla.TDD.Money;

public class Franc
{
    private int amount;

    public Franc(int amount)
    {
        this.amount = amount;
    }

    public Franc Times(int multiplier)
    {
        return new Franc(amount * multiplier);
    }

    public override bool Equals(object? obj)
    {
        return obj is Franc franc && amount == franc.amount;
    }
}

File: java/ch05/src/main/java/com/thesoftwaregorilla/tdd/money/Franc.java
package com.thesoftwaregorilla.tdd.money;

public class Franc {
    private final int amount;

    Franc(int amount) {
        this.amount = amount;
    }

    public Franc times(int multiplier) {
        return new Franc(amount * multiplier);
    }

    public boolean equals(Object object) {
        return object instanceof Franc franc && this.amount == franc.amount;
    }

    public int getAmount() {
        return amount;
    }
}
